include/revng/MFP/MFP.h

```diff
@@ -84,32 +84,14 @@ getMaximalFixedPoint(const MFI &Instance, const typename MFI::GraphType &Flow,
     PartialAnalysis[ExtremalLabel] = ExtremalValue;
   }
 
-  // Handle the special case that the graph has a single entry node
-  if (GT::getEntryNode(Flow) != nullptr) {
-    llvm::ReversePostOrderTraversalExt RPOTE(GT::getEntryNode(Flow), Visited);
-    for (Label Node : RPOTE) {
-      LabelPriority[Node] = LabelPriority.size();
-      Worklist.insert({ LabelPriority.at(Node), Node });
-      // initialize the analysis value for non extremal nodes
-      if (PartialAnalysis.find(Node) == PartialAnalysis.end()) {
-        PartialAnalysis[Node] = InitialValue;
-      }
-    }
-  }
-  // Start visits for nodes that we still haven't visited
-  // prioritizing extremal nodes
-  std::vector<Label> Nodes(ExtremalLabels);
-  for (Label Node : llvm::nodes(Flow)) {
-    Nodes.push_back(Node);
-  }
-  for (Label Start : Nodes) {
```

Comment:
Can you elaborate what this change is doing? I mean, the whole " [WIP][NewMFPABI] Let user specify the entry nodes" commit.

Reply:
This is an issue that arises when we want to use certain `Graphs` from `llvm` as is

For example `GraphTraits<BasisBlock*>` does not implement the complete `GraphTraits` so we cannot iterate over the `nodes` inside that graph without doing a visit first.

Since we already need to do a visit of the graph to get the `RPOT` order of the graph, we only need the user to supply the starting nodes of this visit